pkg/client/client.go

```diff
@@ -461,6 +447,30 @@ func (u *filesUploader) getFilesInfos() ([]fileInfo, error) {
 	return infos, nil
 }
 
+func (u *filesUploader) sendJobs(ctx context.Context, jobsChan chan<- uploadChunkJob, infos []fileInfo) {
+	for _, info := range infos {
+		for i := 0; i < info.TotalChunks; i++ {
+			job := uploadChunkJob{
+				Filename:       info.Name,
+				ChunkNumber:    i + 1,
+				TotalChunks:    info.TotalChunks,
+				ChunkSizeBytes: u.ChunkSizeBytes,
+			}
+			msgSent := false
+			for !msgSent {
+				select {
+				case <-ctx.Done():
+					return
+				case jobsChan <- job:
+					msgSent = true
+				default:
+					continue
+				}
+			}
```

Comment:
The `default` in that `select` clause will cause it to do a "busy wait" for one of the channels to become ready, instead of just sleeping until it happens which would safe a lot of cpu cycles. I'd drop the loop and the boolean and just do
```
select {
case <-ctx.Done():
  return
case jobsChan <- job:
  // nothing
}
```

Reply:
Done